### PR TITLE
Add hint to 'app list' when no apps registered

### PR DESCRIPTION
**Pull request by ericbottard**
Fixes spring-cloud/spring-cloud-dataflow#637

### CODE REVIEW

File: spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/AppRegistryCommands.java [hunk 1]
```diff
@@ -192,6 +191,12 @@ public Table list() {
 			column.add(appRegistration.getName());
 			max = Math.max(max, column.size());
 		}
+		if (max == 0) {
+			return String.format("There are no registered apps at the moment.%n" +
+					"You can register new apps with the '%s' command.", REGISTER_APPLICATION);
```
Comment (markfisher) [inline spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/AppRegistryCommands.java]: should mention `app import` as well?

File: spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/AppRegistryCommands.java [hunk 2]
```diff
@@ -192,6 +191,12 @@ public Table list() {
 			column.add(appRegistration.getName());
 			max = Math.max(max, column.size());
 		}
+		if (max == 0) {
+			return String.format("There are no registered apps at the moment.%n" +
```
Comment (markfisher) [inline spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/AppRegistryCommands.java]: could be more concise, e.g. "No apps registered."

**Comment by ericbottard** [comment]
Addressed review comments

**Comment by markfisher** [comment]
squashed and merged as https://github.com/spring-cloud/spring-cloud-dataflow/commit/682586426c5b8d29d602af7af001933d4841b26b

**Pull request closed without merging**